packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/SdkCapabilityChecker.java

```diff
@@ -0,0 +1,53 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.camera;
+
+import android.annotation.SuppressLint;
+import android.os.Build;
+import androidx.annotation.ChecksSdkIntAtLeast;
+import androidx.annotation.VisibleForTesting;
+
+/** Abstracts SDK version checks, and allows overriding them in unit tests. */
+public class SdkCapabilityChecker {
+  /** The current SDK version, overridable for testing. */
+  @SuppressLint("AnnotateVersionCheck")
+  @VisibleForTesting
+  public static int SDK_VERSION = Build.VERSION.SDK_INT;
+
+  @ChecksSdkIntAtLeast(api = Build.VERSION_CODES.P)
+  public static boolean supportsDistortionCorrection() {
```

Comment:
I find these types of classes aid in readability a ton. That said it is even better when there are breadcrumbs to the source of truth but are annoying to find during a refactor and this is already a net improvement. To that end I have added suggestions with links to the android docs for at many as I could find.